src/app/services/app-auth.service.ts

```diff
@@ -0,0 +1,71 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export type User = {
+  id: string;
+  name: string;
+  balance: number;
+};
+
+/**
+ * This is not the actual implementation of authentication service,
+ * this is only example, be careful with reading all of the notes inside.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class AppAuthService {
+  private isAuthenticated$$ = new BehaviorSubject<boolean>(false);
+
+  private currentUser$$ = new BehaviorSubject<User | null>(null);
```

Comment:
if I refresh the page, I lose my login. We should store that in localStorage/sessionStorage to keep the session alive.